File: Grabid-Android/app/src/main/java/com/grabid/adapters/NotificationAdapter.java
package com.grabid.adapters;

import android.content.Context;
import android.graphics.Typeface;
import android.graphics.drawable.GradientDrawable;
import android.graphics.drawable.LayerDrawable;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.grabid.R;
import com.grabid.models.Notification;

import java.util.ArrayList;

/**
 * Created by vinod on 10/27/2016.
 */
public class NotificationAdapter extends BaseAdapter {
    Context ctx;
    LayoutInflater inflater;
    ArrayList<Notification> notificationData;
    Typeface normal, bold;

    public NotificationAdapter(Context ctx, ArrayList<Notification> notificationData) {
        this.ctx = ctx;
        normal = Typeface.createFromAsset(ctx.getAssets(), "fonts/Raleway_Regular.ttf");
        bold = Typeface.createFromAsset(ctx.getAssets(), "fonts/Raleway_Bold.ttf");
        this.notificationData = notificationData;
        inflater = (LayoutInflater) ctx.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public int getCount() {
        return notificationData.size();
    }

    @Override
    public Object getItem(int i) {
        return notificationData.get(i);
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View mView, ViewGroup viewGroup) {
        ViewHolder viewHolder;
        Notification data = (Notification) getItem(i);
        if (mView == null) {
            mView = inflater.inflate(R.layout.adapter_notification, null);
            viewHolder = new ViewHolder();
            viewHolder.title = (TextView) mView.findViewById(R.id.title);
            viewHolder.message = (TextView) mView.findViewById(R.id.message);
            viewHolder.counter = (TextView) mView.findViewById(R.id.notification_counter);
            viewHolder.card_background = (RelativeLayout) mView.findViewById(R.id.card_layout);
            mView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) mView.getTag();
        }

        LayerDrawable bgDrawable = (LayerDrawable) viewHolder.card_background.getBackground();
        if (data.getReadStatus().equals("0")) {
            //title.setTypeface(bold);
            viewHolder.message.setTypeface(bold);
            /*final GradientDrawable shapeall = (GradientDrawable)   bgDrawable.findDrawableByLayerId(R.id.gradientDrawbleall);
            shapeall.setColor(ContextCompat.getColor(ctx, R.color.notify_read));*/

        } else {
           /* final GradientDrawable shapeall = (GradientDrawable)   bgDrawable.findDrawableByLayerId(R.id.gradientDrawbleall);
            shapeall.setColor(ContextCompat.getColor(ctx, R.color.white));*/
            //title.setTypeface(normal);
            viewHolder.message.setTypeface(normal);
        }
        if (i % 2 == 0) {
            //card_background.setBackgroundResource(R.drawable.colored_blue_sidebar);
            viewHolder.counter.setBackgroundResource(R.drawable.triangle_blue_corner);
            try {
                final GradientDrawable shape = (GradientDrawable) bgDrawable.findDrawableByLayerId(R.id.gradientDrawble);
                shape.setColor(ContextCompat.getColor(ctx, R.color.pending_color));
            } catch (Exception e) {
                e.toString();
            }
        } else if (i % 2 != 0) {
            //card_background.setBackgroundResource(R.drawable.colored_green_sidebar);
            viewHolder.counter.setBackgroundResource(R.drawable.triangle_green_corner);
            final GradientDrawable shape = (GradientDrawable) bgDrawable.findDrawableByLayerId(R.id.gradientDrawble);
            shape.setColor(ContextCompat.getColor(ctx, R.color.completed_color));
        }
        viewHolder.counter.setText("" + (i + 1));
        if (data.getType() != null && (data.getType().contentEquals("33") || data.getType().contentEquals("34") || data.getType().contentEquals("66")))
            viewHolder.title.setText(data.getTitle());
        else
            viewHolder.title.setText(ctx.getResources().getString(R.string.not_title) + " " + data.getDeliveryTitle());
        viewHolder.message.setText(data.getMessage());
        return mView;
    }

    static class ViewHolder {
        TextView title;
        TextView message;
        TextView counter;
        RelativeLayout card_background;
    }
}

File: tipbx/app/src/main/java/com/tipbox/app/adapter/NotificationAdapter.kt
package com.sample.app.adapter

import android.content.Context
import android.graphics.Typeface
import android.graphics.drawable.GradientDrawable
import android.graphics.drawable.LayerDrawable
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.text.style.ForegroundColorSpan
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView
import com.tipbox.app.modle.NotificationModle
import com.tipbox.app.util.CustomTypefaceSpan
import com.tipbox.app.util.SimpleSpanBuilder
import android.graphics.BitmapFactory
import android.graphics.drawable.Drawable
import android.os.Build
import com.bumptech.glide.Glide
import com.bumptech.glide.request.target.SimpleTarget
import com.bumptech.glide.request.transition.Transition
import com.tipbox.app.R


class NotificationAdapter(txtList: ArrayList<NotificationModle>) : RecyclerView.Adapter<NotificationAdapter.MyViewHolder>() {
    inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var name: TextView
        var icon: RelativeLayout
        var card_background: RelativeLayout
        init {
            name = view.findViewById(com.tipbox.app.R.id.name) as TextView
            icon = view.findViewById(com.tipbox.app.R.id.icon) as RelativeLayout

            card_background = view.findViewById<View>(com.tipbox.app.R.id.card_layout) as RelativeLayout
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView = LayoutInflater.from(parent.context)
            .inflate(com.tipbox.app.R.layout.list_adapter_notification, parent, false)
        mcon=parent.context
        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val txt = txtList[position]
        holder.name.text = txt.name

        var name=txt.name.split("gave")[0]
        var priceval=txt.name.split("gave")[1]
        var price=priceval.split(" ")[1]
        var boldTypeface = Typeface.create(holder.name.getTypeface(), Typeface.BOLD)
        var regularTypeface = Typeface.create(holder.name.getTypeface(),Typeface.NORMAL)
        var ssb = SimpleSpanBuilder(name, ForegroundColorSpan(ContextCompat.getColor(mcon, com.tipbox.app.R.color.black))
        ,CustomTypefaceSpan(boldTypeface)
        )
        ssb += SimpleSpanBuilder.Span(
            " gave ",
            ForegroundColorSpan(ContextCompat.getColor(mcon, com.tipbox.app.R.color.nav_text)), CustomTypefaceSpan(regularTypeface)
        )
        ssb += SimpleSpanBuilder.Span(
            price,
            ForegroundColorSpan(ContextCompat.getColor(mcon, com.tipbox.app.R.color.black)),CustomTypefaceSpan(boldTypeface)
        )
        ssb += SimpleSpanBuilder.Span(
            " as tip to you.",
            ForegroundColorSpan(ContextCompat.getColor(mcon, com.tipbox.app.R.color.nav_text)),CustomTypefaceSpan(regularTypeface)
        )

        holder.name.text = ssb.build()
        //holder.name.text=txt.name

        Glide.with(mcon).load(txt.price).into(object : SimpleTarget<Drawable>() {
            override fun onResourceReady(resource: Drawable, transition: Transition<in Drawable>?) {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
                    holder.icon.background = resource
                }
            }
        })
        //txt.price
        val bgDrawable = holder.card_background.getBackground() as LayerDrawable
        if (position % 2 == 0) {
            try {
                val shape = bgDrawable.findDrawableByLayerId(com.tipbox.app.R.id.gradientDrawble) as GradientDrawable
                shape.setColor(ContextCompat.getColor(mcon, com.tipbox.app.R.color.colorPrimary))
            } catch (e: Exception) {
                e.toString()
            }

        } else if (position % 2 != 0) {
            val shape = bgDrawable.findDrawableByLayerId(com.tipbox.app.R.id.gradientDrawble) as GradientDrawable
            shape.setColor(ContextCompat.getColor(mcon, com.tipbox.app.R.color.sidebar_clr))
        }

    }

    override fun getItemCount(): Int {
        return txtList.size
    }

    private val txtList: ArrayList<NotificationModle>
    lateinit var  mcon : Context
    init {
        this.txtList = txtList
    }


}